homeassistant/components/shelly/entity.py

```diff
@@ -370,16 +371,22 @@ def extra_state_attributes(self) -> dict:
 class ShellySleepingBlockAttributeEntity(ShellyBlockAttributeEntity, RestoreEntity):
     """Represent a shelly sleeping block attribute entity."""
 
+    sensors = None
+
     # pylint: disable=super-init-not-called
     def __init__(
         self,
         wrapper: ShellyDeviceWrapper,
         block: aioshelly.Block,
         attribute: str,
         description: BlockAttributeDescription,
-        entry: ConfigEntry | None = None,
+        entry: entity_registry.RegistryEntry | None = None,
+        sensors: set | None = None,
     ) -> None:
         """Initialize the sleeping sensor."""
+        if sensors is not None and ShellySleepingBlockAttributeEntity.sensors is None:
+            ShellySleepingBlockAttributeEntity.sensors = sensors
```

Comment:
You are assigning a static variable inside a constructor. This means this is overridden each time the constructor runs. 

Did you mean `self.sensors` ?

Reply:
I meant to initialize this only once, however I realized that we have two classes which uses different sensors list so it will not work if this is initialized only once. For now each instance will hold this list.